dev-tools/set_version

```diff
@@ -1,38 +1,136 @@
 #!/usr/bin/env python
-import os
 import argparse
+import fileinput
+import os
+import re
+import sys
 from subprocess import check_call
 
-template_go = '''package beat
+pattern = re.compile(r'(const\s|)(\w*)(v|V)(ersion)(\s*)(:|=)(\s)(")(.*)(")(.*)')
+repl    = r'\1\2\3\4\5\6\7"{}"\11'
+
 
-const defaultBeatVersion = "{}"
+goversion_template = '''package main
+
+const appVersion = "{version}"
 '''
 
-template_packer = '''version: "{version}"
+goversion_template_libbeat = '''package beat
+
+const defaultBeatVersion = "{version}"
 '''
 
 
+yamlversion_template = '''version: "{version}"
+'''
+
+def split_path(path):
+    allparts = []
+    while 1:
+        parts = os.path.split(path)
+        if parts[0] == path:  # sentinel for absolute paths
+            allparts.insert(0, parts[0])
+            break
+        elif parts[1] == path: # sentinel for relative paths
+            allparts.insert(0, parts[1])
+            break
+        else:
+            path = parts[0]
+            allparts.insert(0, parts[1])
+    return allparts
+
+def get_substituted_content(filename, pattern, replacement, errormsg):
```

Comment:
The templates are only used to create the files if they don't exist.

When the version file exists, the script just replace the old version number by the new one using a regular expression in `get_substituted_content`. Thus, the community beat can have the version number in the `main.go` instead of `version.go`:

Also, the script is longer due to:

1. detection of libbeat/community beat
2. error handling
3. error messages

```
const myVersion = "0.1.0-alpha2"
func main() {
	err := beat.Run("examplebeat", myVersion, beater.New)
}
```

The regular expression detects for:
`const <prefix>Version = "xxxx"` (go file) or `version: "xxx"` (yml file)